src/components/results/loadflow/load-flow-result.tsx

```diff
@@ -0,0 +1,271 @@
+            //const value = cellData.data[cellData.colDef.field];
+            const value = cellData.value;
+            return (
+                <div className={classes.cell}>
+                    {!isNaN(value) ? value.toFixed(1) : ''}
+                </div>
+            );
+        },
+        [classes.cell]
+    );
+    const loadFlowResultColumns = useMemo(() => {
+        return loadFlowResultColumnsDefinition(
+            intl,
+            StatusCellRender,
+            NumberRenderer
+        );
+    }, [intl, NumberRenderer, StatusCellRender]);
+
+    const messages = useIntlResultStatusMessages(intl);
+
+    useEffect(() => {
+        if (result) {
+            fetchLimitViolations(
+                studyUuid,
+                nodeUuid,
+                limitReductionParam / 100.0
+            )
+                .then((overloadedEquipments: OverloadedEquipmentFromBack[]) => {
+                    const sortedLines = overloadedEquipments
+                        .map((overloadedEquipment) =>
+                            makeData(overloadedEquipment, intl)
+                        )
+                        .sort((a, b) => b.overload - a.overload);
+                    setOverloadedEquipments(sortedLines);
+                })
+                .catch((error) => {
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'ErrFetchViolationsMsg',
+                    });
+                });
+        }
+    }, [studyUuid, nodeUuid, intl, limitReductionParam, result, snackError]);
+
+    const getRowStyle = useCallback(
+        (params: RowClassParams) => {
+            if (params?.data?.elementId) {
+                return {
+                    backgroundColor: theme.selectedRow.background,
+                };
+            }
+        },
+        [theme.selectedRow.background]
+    );
+
+    const onGridReady = useCallback((params: GridReadyEvent) => {
+        if (params.api) {
+            params.api.sizeColumnsToFit();
+        }
+    }, []);
+
+    const currentViolations = overloadedEquipments.filter(
+        (overloadedEquipment) =>
+            overloadedEquipment.limitType === LimitTypes.CURRENT
+    );
+
+    const renderLoadFlowCurrentViolations = () => {
+        const message = getNoRowsMessage(
+            messages,
+            currentViolations,
+            loadFlowStatus
+        );
+        const rowsToShow = getRows(currentViolations, loadFlowStatus);
+        return (
+            <CustomAGGrid
+                rowData={rowsToShow}
+                defaultColDef={defaultColDef}
+                enableCellTextSelection={true}
+                columnDefs={loadFlowCurrentViolationsColumns}
+                onGridReady={onGridReady}
+                getRowStyle={getRowStyle}
+                overlayNoRowsTemplate={message}
+            />
+        );
+    };
+
+    const voltageViolations = overloadedEquipments?.filter(
```

Comment:
**overloadedEquipments** cannot be null or undefined, so there's no need for optional chaining

Reply:
Indeed.